fix(Drawer): initial width doesn't respect min/max widths

kmcfaul opened the pull request:
<!-- What changes are being made? Please link the issue being addressed. -->
**What**: Closes #5278 & #5280

Adds a new `defaultSize` prop to control starting width for resizable drawers. 

A resizable drawer will start at `defaultSize`, or `minSize` if `defaultSize` is undefined and `minSize` is given, or the default drawer width if neither property is given.

Added default and min sizes to the examples to address #5280 (because our examples, which focus the drawer content, was also causing trouble for the scrolling).

patternfly-build commented:
PF4 preview: https://patternfly-react-pr-5313.surge.sh

tlabaj reviewed (changes requested):
Looks good. Can you update the integration app and add a test for this please.

jessiehuff reviewed (approved):
LGTM! :)

kmcfaul commented:
Added default and min sizes for our resizable examples because the code, which focuses the drawer panel content on expansion, was also causing the bug in #5280 by scrolling. 

@mcoker I noticed while adding min sizes that for a left expanding resizable drawer, if the min size is too small, the splitter will disappear into the panel before hitting the min width. I think this is because the splitter is after the panel content (which gets the min width)? If you decrease the left-drawer's `minSize` to `150` from `200`, you can see the behavior.

mcoker commented:
@kmcfaul 

> If you decrease the left-drawer's minSize to 150 from 200, you can see the behavior.

It looks like if we add the `.pf-c-drawer__panel-main` element adjacent to the splitter in resizable drawers, that should fix this problem.

Also for resizable drawers, we need to update the var we use to customize the width from `--pf-c-drawer__panel--FlexBasis` to `--pf-c-drawer__panel--m-resizable--FlexBasis`. We only apply changes to that var if they're above the mobile layout breakpoint.

kmcfaul commented:
For clarity, the changes @mcoker suggests I have ready to go pending another core bump, but these don't impact this PR and can be added in a follow up if necessary. This PR should be ready to go in.

kmcfaul commented on packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:
```diff
@@ -17,8 +17,10 @@ export interface DrawerPanelContentProps extends React.HTMLProps<HTMLDivElement>
-  /** The minimum size of a resizable drawer, in pixels. Defaults to the starting width of the drawer. */
+  /** The minimum size of a resizable drawer, in pixels. Defaults to 0. */
   minSize?: number;
+  /** The starting size of a resizable drawer, in pixels. Defaults to minSize. */
```
Oh nice catch for this bug. The code isn't setting the style because 0 === false for the check to add it. I'll fix the logic.

kmcfaul commented:
Last big update for this PR, pulling in the latest core changes which fixes #5283 as well as updates how min and max size are set and controlled.

mcoker commented on packages/react-core/src/components/Drawer/DrawerPanelContent.tsx:
```diff
@@ -17,17 +17,19 @@ export interface DrawerPanelContentProps extends React.HTMLProps<HTMLDivElement>
   minSize?: number;
-  /** The maximum size of a resizable drawer, in pixels. Defaults to the max width of the parent container. */
+  /** The starting size of a resizable drawer, in pixels. */
+  defaultSize?: number;
```
is there a requirement for this to be in pixels? The var doesn't care what the unit is - px, %, em vw, etc... and that works with the CSS that keeps it within the `minSize`/`maxSize`

kmcfaul replied:
I could change it to a string property. The user would have to specify `px` or `%` in that case. Would that be okay?

kmcfaul replied:
I had it in pixels so I could limit how much the `mousemove` would calculate and set a new size (throwing out the sizes outside the min/max). But since that's being done in CSS I can probably remove them.

kmcfaul commented:
Updated `minSize`, `maxSize`, and `defaultSize` to `string` type, to support % boundaries. Removed code-side checking of boundaries because CSS maintains them. Updated docs.

mcoker commented:
Just a nit but looks like there are 2 "Stacked content body elements" examples. Also I believe the "basic" and "panel on right", and "inline" and "inline panel on right" and are the same?

mcoker reviewed (approved):
LGTM! Great work @kmcfaul!! 🥳

tlabaj commented:
Just need a rebase than it can go in.

evwilkin reviewed (approved):
LGTM

patternfly-build commented:
Your changes have been released in: 
 - @patternfly/react-catalog-view-extension@4.9.10
 - @patternfly/react-charts@6.13.6
 - @patternfly/react-console@4.2.10
 - @patternfly/react-core@4.89.1
 - @patternfly/react-docs@5.14.1
 - @patternfly/react-icons@4.8.3
 - @patternfly/react-inline-edit-extension@4.6.10
 - demo-app-ts@4.68.1
 - @patternfly/react-integration@4.68.1
 - @patternfly/react-styles@4.7.27
 - @patternfly/react-table@4.20.10
 - @patternfly/react-tokens@4.9.25
 - @patternfly/react-topology@4.7.10
 - @patternfly/react-virtualized-extension@4.6.10

Thanks for your contribution! :tada:
